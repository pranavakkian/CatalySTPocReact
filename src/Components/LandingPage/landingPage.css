.addbutton
{
    margin-top: 30px;
}
.viewbutton
{
    margin-top: 30px;
}