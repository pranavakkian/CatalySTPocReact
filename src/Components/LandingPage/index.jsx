import { useDispatch, useSelector } from 'react-redux';
import { memo } from 'react';
import { useHistory } from 'react-router-dom';
import { Button, Container, Header } from 'semantic-ui-react';
import { addColleaguePath } from '../../Constants';
import { handleIsLoggedIn, handleSetUserName } from '../../Redux/UserInfoReducer/Action';
import './landingPage.css';

const LandingPage = memo(() => {
  const history = useHistory();
  const userName = useSelector((state) => state.UserInfoReducer.userName);
  const dispatch = useDispatch();
  const signout = () => {
    dispatch(handleIsLoggedIn(false));
    dispatch(handleSetUserName(null));
  };
  return (
    <div>
      <Button floated="right" type="submit" onClick={signout}>Sign-out</Button>
      <Header as="h1" textAlign="center">
        Welcome
        {' '}
        {userName}
      </Header>
      <Container textAlign="center">
        <div className="addbutton">
          <Button onClick={() => history.push(addColleaguePath)}>Add Colleague</Button>
        </div>
        <div className="viewbutton">
          <Button>View Colleague</Button>
        </div>
      </Container>

    </div>
  );
});

export default LandingPage;
