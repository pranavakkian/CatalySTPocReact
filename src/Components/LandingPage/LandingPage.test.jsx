import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import LandingPage from './index';

describe('LandingPage', () => {
  const mockStore = configureMockStore();

  test('Component rendered successfully', () => {
    const store = mockStore(() => ({
      UserInfoReducer: {
        isLoggedIn: true,
        userName: 'test',
      },
    }));
    const { container } = render(<Provider store={store}><LandingPage /></Provider>);
    expect(container).toMatchSnapshot();
  });
  test('Submit Button Working Properly', () => {
    const store = mockStore(() => ({
      UserInfoReducer: {
        isLoggedIn: true,
        userName: 'test',
      },
    }));
    render(<Provider store={store}><LandingPage /></Provider>);
    userEvent.click(screen.getByRole('button', { name: 'Sign-out' }));
    const expectedAction = [
      { payload: false, type: 'IS_LOGGED_IN' },
      { payload: null, type: 'SET_USER_NAME' },
    ];
    expect(store.getActions()).toStrictEqual(expectedAction);
  });
});
