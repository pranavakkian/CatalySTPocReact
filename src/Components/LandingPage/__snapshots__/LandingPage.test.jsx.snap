// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component rendered successfully 1`] = `
<div>
  <div>
    <button
      class="ui right floated button"
      type="submit"
    >
      Sign-out
    </button>
    <h1
      class="ui center aligned header"
    >
      Welcome
       
      test
    </h1>
    <div
      class="ui center aligned container"
    >
      <div
        class="addbutton"
      >
        <button
          class="ui button"
        >
          Add Colleague
        </button>
      </div>
      <div
        class="viewbutton"
      >
        <button
          class="ui button"
        >
          View Colleague
        </button>
      </div>
    </div>
  </div>
</div>
`;
